File: go/digit-degree/digit-degree.go
package main

import (
	"fmt"
	"strconv"
)

func sumDigits(n int) int{
	nStr := strconv.Itoa(n)
	nSum := 0
	for _, val := range nStr{
		digit, _ := strconv.Atoi(string(val))
		nSum += digit
	}
	return nSum
}

func digitDegree(n int) int {
	if n < 10{
        return 0
	}else{
		return 1 + digitDegree(sumDigits(n))
	}
	
}

func main(){
	n := 91

	fmt.Printf("The digit degree of the number (number of times we need to replace this number with the sum of its digits until we get to a one digit number) %d is: %d", n,  digitDegree(n))
}

File: js/digit-degree.js
function digitDegree(n){
    if (n < 10){
        return 0
    }else{
        return 1 + digitDegree(Number(String(n).split('').map(Number).reduce((a, b) => a + b)))
    }
}

let n = 91
console.log(`The digit degree of the number (number of times we need to replace this number with the sum of its digits until we get to a one digit number) ${n} is:`, digitDegree(n))